### PR TITLE
Registration - add status blocks to firm index page

### PR DESCRIPTION
**Pull request by urfolomeus**
In order to help Principals who are signing up their firms and trading names to know how far through the process they are, we have introduced a status panel against each firm or trading name.

To summarise, each status panel has:

1) An overall status of "Visible on directory" if the firm is published or an overall status of "Needs more information" if more information is still required before it can be published.
2) An indication as to whether all details from that firm have been added with a button to Edit the firm details.
3) An indiction of whether or not the firm needs to add advisers (please note that remote firms do not need advisers which is why some firms/trading names have 0 advisers and yet are ticked), and a button to either Add an adviser (if firm has no advisers) or to Edit the advisers (if the firm has at least one adviser)
4) An indiction of whether or not the firm needs to add offices, and a button to either Add an office (if firm has no offices) or to Edit the offices (if the firm has at least one office)

screenshot 2016-02-23 16 58 23

### CODE REVIEW

**Comment by cazrin** [comment]
:+1:

File: app/presenters/self_service/status_presenter.rb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+    def icon_toggle(condition_met)
+      condition_met ? 'tick' : 'exclamation'
+    end
+
+    def remote?
+      @firm.primary_advice_method == :remote
+    end
+  end
+end
```
Comment (tomas-stefano) [inline app/presenters/self_service/status_presenter.rb]: This is good :+1: Just for the sake of passing knowledge you can inherit from SimpleDelegator and act exactly like a Presenter and proxying the object on initialize.

**Comment by tomas-stefano** [comment]
The ruby part is :+1: But I think that is better other frontend from MAS review this

File: app/presenters/self_service/status_presenter.rb (hunk 1 above)
Comment (urfolomeus) [inline app/presenters/self_service/status_presenter.rb]: Thanks for the info :)

**Comment by munckymagik** [comment]
:+1:

**Comment by johnplayer1982** [comment]
Front end looks good, tests are thorough and @urfolomeus set up a screenshare for a demo, nice! :+1:

**Comment by urfolomeus** [comment]
Thanks @cazrin @tomas-stefano @munckymagik @johnplayer1982 :)

**Pull request merged**